nncf/experimental/torch/nas/bootstrapNAS/training/cosine_lr_scheduler.py

```diff
@@ -0,0 +1,115 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import math
+from typing import Any
+from typing import Dict
+from typing import NoReturn
+from typing import Optional
+
+from nncf.common.schedulers import BaseCompressionScheduler
+from nncf.common.utils.logger import logger as nncf_logger
+
+
+def adjust_learning_rate(optimizer, epoch, init_lr, epochs, batch=0, nBatch=None, lr_schedule_type='cosine'):
+    new_lr = calc_learning_rate(epoch, init_lr, epochs, batch, nBatch, lr_schedule_type)
+    for param_group in optimizer.param_groups:
+        param_group['lr'] = new_lr
+    return new_lr
+
+
+def warmup_adjust_learning_rate(optimizer, init_lr, T_total, nBatch, epoch, batch=0, warmup_lr=0):
+    T_cur = epoch * nBatch + batch + 1
+    new_lr = T_cur / T_total * (init_lr - warmup_lr) + warmup_lr
+    for param_group in optimizer.param_groups:
+        param_group['lr'] = new_lr
+    return new_lr
+
+
+def calc_learning_rate(epoch, init_lr, n_epochs, batch=0, nBatch=None, lr_schedule_type='cosine'):
+    if lr_schedule_type == 'cosine':
+        t_total = n_epochs * nBatch
+        t_cur = epoch * nBatch + batch
+        lr = 0.5 * init_lr * (1 + math.cos(math.pi * t_cur / t_total))
```

Comment:
I've got an error
```
 new_lr = calc_learning_rate(epoch, init_lr, epochs, batch, nBatch, lr_schedule_type)
E             File "/home/nlyalyus/Developer/projects/nn-compression-framework/nncf/experimental/torch/nas/bootstrapNAS/training/cosine_lr_scheduler.py", line 43, in calc_learning_rate
E               lr = 0.5 * init_lr * (1 + math.cos(math.pi * t_cur / t_total))
E           TypeError: unsupported operand type(s) for *: 'float' and 'NoneType'
```

Reply:
Fixed. I introduced a typo by mistake in a previous commit. Thanks!